test/e2e/framework/framework.go

```diff
@@ -450,10 +474,19 @@ func convertAzureProviderConfigToControlPlaneMachineSetProviderSpec(providerConf
 
 // convertVSphereProviderConfigToControlPlaneMachineSetProviderSpec converts an VSphere providerConfig into a
 // raw control plane machine set provider spec.
-func convertVSphereProviderConfigToControlPlaneMachineSetProviderSpec(providerConfig providerconfig.ProviderConfig) (*runtime.RawExtension, error) {
+func convertVSphereProviderConfigToControlPlaneMachineSetProviderSpec(providerConfig providerconfig.ProviderConfig, hasFailureDomains bool) (*runtime.RawExtension, error) {
 	vspherePs := providerConfig.VSphere().Config()
+	vspherePs = *vspherePs.DeepCopy()
 	vspherePs.Name = ""
 
+	// vSphere handles failure domains differently.  There will not be a 1-to-1 match for
+	// new machine vs cpms.  Need to strip
```

Comment:
can you expand on this comment?

Reply:
This is in relation to how the matching logic works in the cpmso.  It does a direct == compare, but we relly only care about a fe w fields to diff against..  I can remove this comment if its confusing.